### PR TITLE
feat: Create Track in the app

### LINKED ISSUE #871: Allow creation of Tracks in the app
### Feature Request

**Feature Description**

The app should support creation of Tracks. I propose to implement it in a similar way to the Creation of an Event.

**Would you like to work on the issue?**

Yes

**Parent issue**
#870 

To be done after #872

**Update**

~~~Busy till 13th April 2018(Friday) hence can be taken by someone else before that, Otherwise i will work on it on the weekend.~~~
Working on it!

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #871 

Changes: Create Track in the app

Screenshots for the change: 
videotogif_2018 04 14_01 18 12

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    private void nullifyEmptyFields(Track track) {
+        track.setDescription(StringUtils.emptyToNull(track.getDescription()));
+    }
+
+    public void createTrack() {
+        nullifyEmptyFields(track);
+
+        track.setEvent(ContextManager.getSelectedEvent());
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java]: See the infinite recursion fix

File: app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackFragment.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private Validator validator;
+
+    public static CreateTrackFragment newInstance() {
+        return new CreateTrackFragment();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        final Context contextThemeWrapper = new ContextThemeWrapper(getActivity(), R.style.AppTheme);
+        LayoutInflater localInflater = inflater.cloneInContext(contextThemeWrapper);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackFragment.java]: Redundant

File: app/src/main/res/drawable/ic_color_lens.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#FF000000"
```
Comment (iamareebjamal) [inline app/src/main/res/drawable/ic_color_lens.xml]: grey

File: app/src/main/res/drawable/ic_format_color_text.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+        android:width="24dp"
+        android:height="24dp"
+        android:viewportWidth="24.0"
+        android:viewportHeight="24.0">
+    <path
+        android:fillColor="#FF000000"
```
Comment (iamareebjamal) [inline app/src/main/res/drawable/ic_format_color_text.xml]: grey

File: app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java (hunk 1 above)
Comment (srv-twry) [reply app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java]: @iamareebjamal 
I have made the required changes but why is setting the selected event on the track/copyright is causing infinite recursion ?

**Pull request merged**